sim/src/make/activity_model.rs

```diff
@@ -2,114 +2,241 @@ use crate::{
+        let mut num_bldg_commercial = 0;
+        let mut num_bldg_mixed_residential_commercial = 0;
         for b in map.all_buildings() {
             match b.bldg_type {
                 BuildingType::Residential(num_ppl) => {
+                    // TODO is it expected that sometimes num_ppl is 0?
+                    // TODO better estimate, maybe based on parking, buildling size, etc
+                    let num_ppl = if num_ppl > 0 {
+                        num_ppl
+                    } else {
+                        rng.gen_range(1, 5)
+                    };
+
+                    assert!(num_ppl > 0);
                     for _ in 0..num_ppl {
-                        residences.push((b.id, abstutil::fork_rng(rng)));
+                        residents.push(b.id);
                     }
-                    num_residences += 1;
+                    num_bldg_residential += 1;
                 }
                 BuildingType::ResidentialCommercial(num_ppl) => {
+                    assert!(num_ppl > 0);
                     for _ in 0..num_ppl {
-                        residences.push((b.id, abstutil::fork_rng(rng)));
+                        residents.push(b.id);
+
+                        // TODO: currently we just assume work capacity is the same as residential
+                        // capacity, which is surely not true. How can we better estimate?
+                        workers.push(b.id);
                     }
-                    workplaces.push(b.id);
-                    num_residences += 1;
+                    num_bldg_mixed_residential_commercial += 1;
                 }
                 BuildingType::Commercial => {
-                    workplaces.push(b.id);
+                    // TODO: currently we just assume some constant for jobs per building.
+                    // A better metrics might be available parking or building size
+                    let building_cap = usize::pow(2, rng.gen_range(0, 7));
+                    for _ in 0..building_cap {
+                        workers.push(b.id);
+                    }
+                    num_bldg_commercial += 1;
                 }
                 BuildingType::Empty => {}
             }
         }
 
+        residents.shuffle(rng);
+        workers.shuffle(rng);
+
         let mut s = Scenario::empty(map, "random people going to/from work");
         // Include all buses/trains
         s.only_seed_buses = None;
 
-        for mut person in timer
-            .parallelize(
-                "create people",
-                Parallelism::Fastest,
-                residences,
-                |(home, mut forked_rng)| robot_person(home, &workplaces, map, &mut forked_rng),
-            )
-            .into_iter()
-            .flatten()
-        {
-            person.id = PersonID(s.people.len());
-            s.people.push(person);
-        }
+        let residents_cap = residents.len();
+        let workers_cap = workers.len();
```

Comment:
temporarily I've taken out the parallelize, but I'll address it before merging.

Reply:
added back the parallelism, plus a couple other perf things in: https://github.com/dabreegster/abstreet/pull/320/commits/35bc6d4dd8c776d1771e2975074aafaa2eb18a34


Not sure how useful these screenshots will be but you can compare the overall sample count for the loading screen:

**loading before**:
<img width="1792" alt="Screen Shot 2020-09-10 at 12 19 07 PM" src="https://user-images.githubusercontent.com/217057/92789691-efde7e00-f35f-11ea-9e4f-8f17ae388731.png">

**loading after**:
<img width="1792" alt="Screen Shot 2020-09-10 at 12 19 11 PM" src="https://user-images.githubusercontent.com/217057/92789685-ef45e780-f35f-11ea-9beb-fa24ba63ce57.png">

**zoomed into proletariat robot before**
You can see time is in the path-finding, cloning boundary intersections, and lots of logging
<img width="1792" alt="Screen Shot 2020-09-10 at 12 19 34 PM" src="https://user-images.githubusercontent.com/217057/92789666-eb19ca00-f35f-11ea-86ff-a86074e402ec.png">

**zoomed out after**
Got rid of most of the logging, moved cloning boundary outside of loop, and moved the bulk of the remaining path-finding work has been moved to another thread.
<img width="1792" alt="Screen Shot 2020-09-10 at 12 18 22 PM" src="https://user-images.githubusercontent.com/217057/92789694-f0771480-f35f-11ea-95c4-ce725aeda12d.png">

Building the PersonSpec's for my test map is about 3s, compared to 45s of "finding parking". 

Finding parking is by far the biggest chunk of time in building the scenario right now, but that's not new to this PR. I looked briefly and it's going to be trickier to parallelize, since each computation removes a spot from consideration, which might affect concurrent computations.

Reply:
> Finding parking is by far the biggest chunk of time in building the scenario right now

Definitely not clear how to parallelize it, or speed it up in general. One idea I've toyed with is inverting the supply/demand order. Right now we iterate through each item in the demand, looking for supply to fill it. Some of the BFS search work could definitely be reused in some way. What about the other order -- shuffle all available spots, then look for demand to fill each? Potentially it's easier to do that search. More generally, this is some kind of bipartite(ish -- might have unequal number of spots/cars) graph matching problem, with a cost for each car/spot pair. I haven't looked around, but I bet there's some better general approach.